### PR TITLE
LF added

### PR DESCRIPTION
**Pull request by EhsanKhodadad**
In this commit,  lf-workspace folder is added to the apps folder.

### CODE REVIEW

File: c/apps/lf-workspace/hello/src/platform/lf_patmos_support.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#define PRINTF_MICROSTEP "%" PRIu32
+#define PRINTF_TAG "(%" PRId64 ", %" PRIu32 ")"
+
+// Keep track of physical actions being entered into the system
+static volatile bool _lf_async_event = false;
+// Keep track of nested critical sections
+static uint32_t _lf_num_nested_critical_sections=0;
+/**
+ * Global timing variables:
+ * Since Arduino is 32bit, we need to also maintain the 32 higher bits.
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/platform/lf_patmos_support.h:60]: Update comments to fit Patmos

File: c/apps/lf-workspace/hello/src/platform/lf_patmos_support.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+/**
+ * FlexPRET API support for the C target of Lingua Franca.
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/platform/lf_patmos_support.h:29]: Update to fit Patmos

File: c/apps/lf-workspace/hello/src/scripts/Commands.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+lfc c/apps/lf-workspace/hello/src/HelloWorld.lf
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/scripts/Commands.txt:1]: These commands should be accompanied by desctitions about what they are intended to do.

File: c/apps/lf-workspace/hello/src/scripts/patmos_build.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/scripts/patmos_build.sh:1]: Add comments about what this does and how to use it

File: c/apps/lf-workspace/hello/src/Makefile [hunk 5]
```diff
@@ -0,0 +1,38 @@
+LF_PROJECT_ROOT := "/home/ehsan/t-crest/patmos/c/apps/lf-workspace/hello/"
+LF_MAIN_TARGET := HelloWorld
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/Makefile:2]: Does this hardcode the program to be naked `HelloWorld`. Shouldnt you be able to use another name?

File: c/apps/lf-workspace/hello/src/Makefile [hunk 6]
```diff
@@ -0,0 +1,38 @@
+LF_PROJECT_ROOT := "/home/ehsan/t-crest/patmos/c/apps/lf-workspace/hello/"
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/Makefile:1]: This path is hardcoded to Ehsan's machine, so will fail on any other. Should use relative paths

File: c/apps/lf-workspace/hello/src/Makefile [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+INCS  := -I"$(LF_PROJECT_ROOT)/include" \
+    -I"$(LF_PROJECT_ROOT)/include/api" \
+    -I"$(LF_PROJECT_ROOT)/include/core" \
+    -I"$(LF_PROJECT_ROOT)/include/core/platform" \
+    -I"$(LF_PROJECT_ROOT)/include/core/modal_models" \
+    -I"$(LF_PROJECT_ROOT)/include/core/utils" 
+CC := patmos-clang 
+CFLAGS := $(INCS) -DINITIAL_EVENT_QUEUE_SIZE=10 -DINITIAL_REACT_QUEUE_SIZE=10 -DLF_UNTHREADED=1# -DLF_REACTION_GRAPH_BREADTH=1 -Wno-everything 
+#CFLAGS += -O2 -mserialize=lf.pml
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/Makefile:13]: Remove unused code in comments

File: c/apps/lf-workspace/hello/src/Makefile [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+SRC_FILES := _helloworld_main.c HelloWorld.c lib/schedule.c _counter.c _double.c _printer.c
```
Comment (Emoun) [inline c/apps/lf-workspace/hello/src/Makefile:14]: hardcoded names, should be fixed

File: c/apps/lf-workspace/hello/src/README.md [hunk 9]
```diff
@@ -0,0 +1,2 @@
+run lfc HelloWorld.lf in src directory, so src-gen, include, and bin folders will be generated. Then run patmos_build.sh to copy patmos support files to the generated folder. finally run make lf-app APP=HelloWorld comp in Patmos folder to comple lf codes.
```
[the remark this answers is not in the document]
Comment (schoeberl) [reply c/apps/lf-workspace/hello/src/README.md:1]: This whole commit should only be a temporary one. The correct integration is Patmos into LF (reactor-c). Not this way around. We should aim for this very soon.

**Comment by schoeberl** [comment]
Please fix all hard-coded paths as Emad suggested. Maybe also the minor ones. As stated above, the "correct" integration of Patmos with LF is in the LF codebase itself (like the Arduino support).

**Comment by Emoun** [comment]
I agree that upstreaming the integration to the LF repository is a good idea. We could then add commands that easily pull in LF and sets it up for use with patmos.

**Pull request merged**